testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/servlet/BrokerLinkAndTokenExchangeTest.java

```diff
@@ -736,6 +721,50 @@ private void navigateTo(String uri) {
         WaitUtils.waitForPageToLoad();
     }
 
-    
+    private void checkFeature(int statusCode) throws Exception {
+        String accessToken = oauth.doGrantAccessTokenRequest(PARENT_IDP, PARENT2_USERNAME, "password", null, PARENT_CLIENT, "password").getAccessToken();
+
+        if (statusCode != Response.Status.NOT_IMPLEMENTED.getStatusCode()) {
+            Assert.assertEquals(0, adminClient.realm(CHILD_IDP).getClientSessionStats().size());
+        }
+
+        Client httpClient = ClientBuilder.newClient();
```

Comment:
Closing the client on line 766 would mean that client is closed just if condition on line 750 (if (statusCode != Response.Status.NOT_IMPLEMENTED.getStatusCode())) is true. Maybe will be good to replace with the auto-closeable try block? Something like "try (Client httpClient = ClientBuilder.newClient()) {" . That should ensure client always closed.

Reply:
Thanks Marek! Unfortunately, I can't use the try-with-resources statement, because `Client` doesn't implement `Closeable` interface and it's necessary. So I'll just use try-finally.

Reply:
That's fine. Thanks!